src/pkg/services/m365/custom/drive_item.go

```diff
@@ -0,0 +1,339 @@
+//
+//nolint:revive
+package custom
+
+import (
+	"strings"
+	"time"
+
+	"github.com/microsoftgraph/msgraph-sdk-go/models"
+
+	"github.com/alcionai/corso/src/internal/common/ptr"
+	"github.com/alcionai/corso/src/internal/common/str"
+)
+
+// Replica of models.DriveItemable
+type LiteDriveItemable interface {
+	GetId() *string
+	GetName() *string
+	GetSize() *int64
+	GetFolder() interface{}
+	GetPackageEscaped() interface{}
+	GetShared() interface{}
+	GetMalware() malwareable
+	GetDeleted() interface{}
+	GetRoot() interface{}
+	GetFile() fileItemable
+	GetParentReference() itemReferenceable
+	SetParentReference(itemReferenceable)
+	GetCreatedBy() identitySetable
+	GetCreatedByUser() userable
+	GetLastModifiedByUser() userable
+	GetCreatedDateTime() *time.Time
+	GetLastModifiedDateTime() *time.Time
+	GetAdditionalData() map[string]interface{}
+}
+
+var _ LiteDriveItemable = &driveItem{}
+
+type driveItem struct {
+	id                   string
+	name                 string
+	size                 int64
+	folder               interface{}
+	pkg                  interface{}
+	shared               interface{}
+	deleted              interface{}
+	root                 interface{}
+	malware              malwareable
+	file                 fileItemable
+	parentRef            itemReferenceable
+	createdBy            identitySetable
+	createdByUser        userable
+	lastModifiedByUser   userable
+	createdDateTime      time.Time
+	lastModifiedDateTime time.Time
+	additionalData       map[string]interface{}
+}
+
+func (c *driveItem) GetId() *string {
+	return &c.id
+}
+
+func (c *driveItem) GetName() *string {
+	return &c.name
+}
+
+func (c *driveItem) GetSize() *int64 {
+	return &c.size
+}
+
+func (c *driveItem) GetFolder() interface{} {
+	return c.folder
+}
+
+func (c *driveItem) GetPackageEscaped() interface{} {
+	return c.pkg
+}
+
+func (c *driveItem) GetShared() interface{} {
+	return c.shared
+}
+
+func (c *driveItem) GetMalware() malwareable {
+	return c.malware
+}
+
+func (c *driveItem) GetDeleted() interface{} {
+	return c.deleted
+}
+
+func (c *driveItem) GetRoot() interface{} {
+	return c.root
+}
+
+func (c *driveItem) GetFile() fileItemable {
+	return c.file
+}
+
+func (c *driveItem) GetParentReference() itemReferenceable {
+	return c.parentRef
+}
+
+// TODO(pandeyabs): Should we only support GETs?
+func (c *driveItem) SetParentReference(parent itemReferenceable) {
+	c.parentRef = parent
+}
+
+func (c *driveItem) GetCreatedBy() identitySetable {
+	return c.createdBy
+}
+
+func (c *driveItem) GetCreatedByUser() userable {
+	return c.createdByUser
+}
+
+func (c *driveItem) GetLastModifiedByUser() userable {
+	return c.lastModifiedByUser
+}
+
+func (c *driveItem) GetCreatedDateTime() *time.Time {
+	return &c.createdDateTime
+}
+
+func (c *driveItem) GetLastModifiedDateTime() *time.Time {
+	return &c.lastModifiedDateTime
+}
+
+func (c *driveItem) GetAdditionalData() map[string]interface{} {
+	return c.additionalData
+}
+
```

Comment:
Comment separators would add a lot of readability between structures here.
```suggestion

// ---------------------------------------------------------------------------
// malwareable
// ---------------------------------------------------------------------------
```